### ISSUE TITLE
Incorrect printing of parse_latex('\\frac {-1 - 1}{2}')

### ISSUE THREAD

**Issue by costika1234** [sympy/sympy#24108]
I've just uncovered a bug when displaying LaTeX expressions involving numerators of the form `-X - Y` or `-X + func(Y)` in combination with purely numerical denominators.

Here's a relevant code snippet that demonstrates the problem: 
```
from sympy.parsing.latex import parse_latex
from sympy import srepr

expr = parse_latex('\\frac {-1 - 1}{2}')

print(expr)
print(srepr(expr))
```

Under `sympy 1.11.1`, the code above produces the following output:
```
-1 - 1*1/2
Mul(Pow(Integer(2), Integer(-1)), Add(Integer(-1), Mul(Integer(-1), Integer(1))))
```

This is clearly a printing problem as the string representation is missing proper parentheses around the `-1 - 1*1` numerator. (The `srepr` is fine, though.)

From what I can tell, the printing seems to fail when we pass `evaluate=False` to `sympy.Mul` in combination with certain arguments. For example, the above is equivalent to:
```
sympy.Mul(sympy.S('-1 - 1', evaluate=False), sympy.Pow(2, -1, evaluate=False), evaluate=False)
```
which again displays the same incorrect string, i.e. `-1 - 1*1/2`.

However, if we change the argument of `sympy.Pow` (i.e. we change the original denominator of the fraction) to an expression like `x`, then the printing is absolutely fine. Indeed, 
```
sympy.Mul(sympy.S('-1 - 1', evaluate=False), sympy.Pow(sympy.S('x'), -1, evaluate=False), evaluate=False)
``` 
displays `(-1 - 1*1)/x`, as desired.

Note that the same bug happens when we replace the numerator with expressions like `-1 + \\sqrt{3}`.

Lastly, here is a list of correct/incorrect string representations that might be helpful during debugging:
```
print(parse_latex('\\frac {-1 + 1}{2}'))
(-1 + 1)/2 ✅

print(parse_latex('\\frac {-1 - 1}{2}'))
-1 - 1*1/2 ❌
```

```
print(parse_latex('\\frac {-1 - 1}{200 + 0}'))
(-1 - 1*1)/(0 + 200) ✅

print(parse_latex('\\frac {-1 - 1}{200}'))
-1 - 1*1/200 ❌
```

```
print(parse_latex('\\frac {0 -x - y}{2}'))
(-y - (x + 0))/2 ✅

print(parse_latex('\\frac {-x - y}{2}'))
-x - y/2 ❌
```

```
print(parse_latex('\\frac {x + \\sqrt{y}}{2}'))
(x + sqrt(y))/2 ✅

print(parse_latex('\\frac {-x + \\sqrt{y}}{2}'))
-x + sqrt(y)/2 ❌
```

**Comment by asmeurer** [comment]
Probably the same issue as https://github.com/sympy/sympy/issues/24022, although I haven't confirmed.

**Comment by costika1234** [comment]
It's not the same as the issue you linked to. That one is related to `sympy.Add`, whereas this bug has to do with `sympy.Mul` instead. I even tried the suggested fix from that discussion and it doesn't help here.

**Comment by oscargus** [comment]
It is impossible to get sensible unevaluated printing at the same time as getting "nice" printing. I cannot really see the fundamental problem here though. 

The parenthesis is removed here:
https://github.com/sympy/sympy/blob/2ba681e0b469c845d944cb1cdcb6008ab56338da/sympy/printing/str.py#L289-L291

Which was introduced in https://github.com/sympy/sympy/pull/21390 (and later on modified to support other unevaluated printing...).

I've fixed this in #24111 by making sure that the term is not an `Add`. There may be other expressions where it is bad to do this as well, but this is the one I could think about. Now it gives `(-1 - 1*1)/2` and that is the best we can do as there is no `Sub` operation and we could not distinguish between the current case and the case where someone actually enters that (which I am sure someone has complained about earlier, although it is a bit strange why this happens as e.g. `parse_latex('\\frac {-1 - 1*1}{2}')` prints as `(-1 - 1)/2`, but that seems to be a consequence of the exact srepr from the parser which differs). 

Will be interesting to see what other unevaluated printing this breaks... None that is tested for if nothing else.

**Comment by asmeurer** [comment]
It would really help if objects created with evaluate=False had some flag indicating that they have skipped canonicalization. Then such objects could dispatch to a completely separate printer, which only does naive structural printing. 

Even so, I'm unsure about the use of could_extract_minus_sign in the printers. That method is supposed to be a way to determine whether one should have `(x - y)` or `-(y - x)`. But in the printers, I don't think it should ever "extract minus sign". It should just print the expression as it is. 

Also, why is the printer calling `str`? The printer should always recursively call `self._print`.
